cni/network/network.go

```diff
@@ -793,28 +808,37 @@ func (plugin *NetPlugin) createEndpointInternal(opt *createEndpointInternalOpt)
 
 	epInfos := []*network.EndpointInfo{&epInfo}
 	// get secondary interface info
-	for _, secondaryCniResult := range opt.ipamAddResult.secondaryInterfacesInfo {
-		var addresses []net.IPNet
-		for _, ipconfig := range secondaryCniResult.IPConfigs {
-			addresses = append(addresses, ipconfig.Address)
-		}
+	for _, secondaryCniResult := range opt.ipamAddResult.interfaceInfo {
+		switch secondaryCniResult.NICType {
+		case cns.DelegatedVMNIC:
+			// secondary
+			var addresses []net.IPNet
+			for _, ipconfig := range secondaryCniResult.IPConfigs {
+				addresses = append(addresses, ipconfig.Address)
+			}
 
-		epInfos = append(epInfos,
-			&network.EndpointInfo{
-				ContainerID:       epInfo.ContainerID,
-				NetNsPath:         epInfo.NetNsPath,
-				IPAddresses:       addresses,
-				Routes:            secondaryCniResult.Routes,
-				MacAddress:        secondaryCniResult.MacAddress,
-				NICType:           secondaryCniResult.NICType,
-				SkipDefaultRoutes: secondaryCniResult.SkipDefaultRoutes,
-			})
+			epInfos = append(epInfos,
+				&network.EndpointInfo{
+					ContainerID:       epInfo.ContainerID,
+					NetNsPath:         epInfo.NetNsPath,
+					IPAddresses:       addresses,
+					Routes:            secondaryCniResult.Routes,
+					MacAddress:        secondaryCniResult.MacAddress,
+					NICType:           secondaryCniResult.NICType,
+					SkipDefaultRoutes: secondaryCniResult.SkipDefaultRoutes,
+				})
+		case cns.BackendNIC:
+			// todo
+		default:
+			// InfraNic
```

Comment:
reassign `defaultIndex` here. the index for `interfaceInfo` and `epInfo` could potentially be different. 

Thanks @QxBytes